src/components/password-modal/password-modal.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { View, Modal, TouchableHighlight, TextInput, Platform } from 'react-native';
+import { translate } from '../../core/i18n';
+import Icon from '../icon';
+import { withTheme } from '../../core/theme/with-theme';
+import stylesProvider from './styles';
+import { Button, Text } from '../../library';
+import { HeaderLeft } from '../header-left/header-left';
+import { ITheme } from '../../core/theme/itheme';
+
+const STATUSBAR_HEIGHT = Platform.select({
+    ios: 20,
+    default: 0
+});
+
+export interface IProps {
+    styles: ReturnType<typeof stylesProvider>;
+    theme: ITheme;
+    visible: boolean;
+    infoText: string;
+    buttonLabel: string;
+    onReject: any;
+    onConfirm: any;
+    obRef: any;
+}
+
+interface IState {
+    password: string;
+    revealPassword: boolean;
+}
+
+export class PasswordModalComponent extends React.Component<IProps, IState> {
+    public onReject: any;
+    public onConfirm: any;
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            password: '',
+            revealPassword: false
+        };
+
+        props.onReject && (this.onReject = props.onReject);
+        props.onConfirm && (this.onConfirm = props.onConfirm);
+
+        props.obRef && props.obRef(this);
+    }
+
+    public requestPassword(callback) {
+        this.onConfirm = callback;
```

Comment:
I would make this more simple. Right now if someone uses `onReject` and `onConfirm` props and also calls `requestPassword` `this.onConfirm` is overwritten. I would loose the callback, and i would return a promise:
```ts
    private passwordRequestDeferred;
    public requestPassword(): Promise<string> {
        this.passwordRequestDeferred = new Deferred();
        this.setState({visible: true});
        return this.passwordRequestDeferred.promise;
    }
```
you can find implementation for Deferred here (https://github.com/cryptolandtech/moonlet/blob/master/src/utils/deferred.ts)

on button presses you should add
```ts
// for reject
this.passwordRequestDeferred && this.passwordRequestDeferred.reject();
// for confirm
this.passwordRequestDeferred && this.passwordRequestDeferred.resolve('PASSWORD');
```

Reply:
i would also auto close modal `onReject` and `onConfirm`